formats/gltf/writer.go

```diff
@@ -317,6 +329,134 @@ func rgbToFloatArr(c color.Color) [3]float64 {
 	}
 }
 
+func (w *Writer) AddScene(scene PolyformScene) error {
+	for _, model := range scene.Models {
+		meshIndex, err := w.AddMesh(model)
+		if err != nil {
+			return fmt.Errorf("failed to add model %q: %w", model.Name, err)
+		}
+
+		// Create node with transforms for this model
+		nodeIndex := len(w.nodes)
+		newNode := Node{
+			Mesh: &meshIndex,
+			Name: model.Name,
+		}
+
+		if model.Translation != nil {
+			arr := model.Translation.ToFixedArr()
+			newNode.Translation = &arr
+		}
+
+		if model.Quaternion != nil {
+			arr := model.Quaternion.ToArr()
+			newNode.Rotation = &arr
+		}
+
+		if model.Scale != nil {
+			arr := model.Scale.ToFixedArr()
+			newNode.Scale = &arr
+		}
+
+		w.nodes = append(w.nodes, newNode)
+		w.scene = append(w.scene, nodeIndex)
+
+		skinNode := nodeIndex
+		// Handle any skeleton/animation data
+		if model.Skeleton != nil {
+			var skinIndex *int
+			skinIndex, skinNode = w.AddSkin(*model.Skeleton)
+			w.nodes[nodeIndex].Skin = skinIndex
+		}
+
+		if len(model.Animations) > 0 {
+			w.AddAnimations(model.Animations, *model.Skeleton, skinNode)
+		}
+	}
+
+	// Add lights
+	for _, light := range scene.Lights {
+		w.AddLight(light)
+	}
+
+	return nil
+}
+
+func (w *Writer) AddMesh(model PolyformModel) (_ int, err error) {
+	if model.Mesh == nil {
+		return -1, fmt.Errorf("nil mesh in model %q", model.Name)
+	}
+
+	// Check for empty mesh
+	if model.Mesh.PrimitiveCount() == 0 {
```

Comment:
There is a test that confirms that empty mesh is correctly rendered into an empty file. 

Shall I preserve that behaviour? Right now in this commit it errors out on empty meshes, but I can revert to previous easily.

Reply:
I've def written pipelines in the past where it's valid to generate an empty GLTF (prob where the test came from lol). Let's keep the behavior and not error out.